lib/sda/local.go

```diff
@@ -173,7 +179,11 @@ func (l *LocalTest) NewNode(tn *TreeNode, protName string) (*Node, error) {
 		TreeNodeID:   tn.Id,
 		RoundID:      uuid.NewV4(),
 	}
-	return NewNode(o, tok)
+	node, err := NewNode(o, tok)
+	if err == nil {
+		l.Nodes = append(l.Nodes, node)
+	}
+	return node, err
```

Comment:
Should we not register the node also in the overlay ? Will it not bring some issues if we don't ? Like some messages not going to the right P.I. because the Overlay did not know about the right Node ?